spellcheck.pl

```diff
@@ -135,20 +137,27 @@ sub spellcheck_key_pressed
 
     return unless Irssi::settings_get_bool('spellcheck_enabled');
 
-    # I know no way to *mark* misspelled words in the input line,
-    # that's why there's no spellcheck_print_suggestions -
-    # because printing suggestions is our only choice.
-
     # hide correction window when message is sent
     if (chr($key) =~ /\A[\r\n]\z/ && $correction_window) {
         $correction_window->command("^window hide $window_name");
+        if (Irssi->can('gui_input_clear_extents')) {
+            Irssi::gui_input_clear_extents(0, 9999);
+        }
+    }
+
+    # get current inputline
+    my $inputline = Irssi::parse_special('$L');
+    if (lc Irssi::settings_get_str('term_charset') eq 'utf-8') {
+        Encode::_utf8_on($inputline);
+    }
+    if (Irssi->can('gui_input_set_extent')) {
+        Irssi::gui_input_set_extent(length $inputline, '%n');
```

Comment:
What was this supposed to do?
It seems to make sure that when you correct an already-colored word, the newly added letters are not colored. But this only works at the end of the input line.

Reply:
good point. you need to ensure that the line ends with %n always, otherwise if you delete the last character the underline will go to the end of the input prompt. but in fact the same issue will happen if you delete the last character of an underlined word :thinking: